apps/spruce/src/pages/adminSettings/AdminSaveButton.tsx

```diff
@@ -0,0 +1,58 @@
+import { useMutation } from "@apollo/client";
+import Button from "@leafygreen-ui/button";
+import {
+  SaveAdminSettingsMutation,
+  SaveAdminSettingsMutationVariables,
+} from "gql/generated/types";
+import { SAVE_ADMIN_SETTINGS } from "gql/mutations";
+import { useAdminSettingsContext } from "./Context";
+import { formToGqlMap } from "./tabs/transformers";
+import { FormToGqlFunction } from "./tabs/types";
+
+export const AdminSaveButton = () => {
+  const { checkHasUnsavedChanges, getChangedTabs, getTab } =
+    useAdminSettingsContext();
+  const hasUnsavedChanges = checkHasUnsavedChanges();
+
+  const [saveAdminSettings] = useMutation<
+    SaveAdminSettingsMutation,
+    SaveAdminSettingsMutationVariables
+  >(SAVE_ADMIN_SETTINGS, {});
+
+  const handleSave = () => {
+    const changedTabs = getChangedTabs();
+    console.log("Saving changes...", changedTabs);
+
+    const changedSettings = changedTabs.reduce(
+      (acc, tab) => {
+        if (Object.prototype.hasOwnProperty.call(formToGqlMap, tab)) {
+          const { formData } = getTab(tab);
+          // @ts-expect-error: FIXME. This comment was added by an automated script.
+          const formToGql: FormToGqlFunction<typeof tab> = formToGqlMap[tab];
+          const changes = formToGql(formData);
+          return { ...acc, ...changes };
```

Comment:
```suggestion
      const formToGqlFn: FormToGqlFunction<typeof tab> = formToGqlMap[tab];
      if (formToGqlFn) {
        const { formData } = getTab(tab);
        const changes = formToGqlFn(formData);
        return { ...acc, ...changes };
```
maybe something like this, you may have to fiddle with types (but just do `ts-expect-error`s if it's not working out)

Reply:
I spent a while on it, I can't seem to get the types to work i'm sorry